<?xml version="1.0" ?>

<robot name="UR3_with_k4a" xmlns:xacro="http://ros.org/wiki/xacro">
  <xacro:include filename="$(find robotiq_2f_85_gripper_visualization)/urdf/robotiq_arg2f_85_model_macro.xacro" />
  <xacro:robotiq_arg2f_85 prefix=""/>

  <xacro:include filename="$(find ur_description)/urdf/ur3.urdf.xacro"/>
  <xacro:ur3_robot prefix="" joint_limited="false"/>

  <link name="world"/>
  <link name="winml_link"/>
  <link name="k4a_link_optical" />

  <link name="table">
      <visual>
          <geometry>
              <box size="1.0 1.0 0.012"/>
          </geometry>
          <material name="Green">
          <color rgba="0.0 0.5 0.0 0.5"/>
          </material>
      </visual>
      <collision>
          <geometry>
              <box size="1.0 1.0 0.012"/>
          </geometry>
      </collision>
  </link>

  <link name="robot_mount">
    <visual>
      <geometry>
        <cylinder length="0.012" radius="0.0725"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.012" radius="0.0725"/>
      </geometry>
    </collision>
  </link>

  <link name="k_mount" >
    <visual>
      <geometry>
        <mesh filename="package://k4a_arm_support/meshes/lifecam_mount.dae"/>
      </geometry>
      <material name="Blue">
        <color rgba="0.0 0.0 0.5 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://k4a_arm_support/meshes/lifecam_mount.dae"/>
      </geometry>
    </collision>
  </link>

  <link name="k4a" >
    <visual>
      <geometry>
        <cylinder length="0.07" radius="0.02"/>
      </geometry>
      <material name="White">
        <color rgba="0.9 0.9 0.9 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.07" radius="0.015"/>
      </geometry>
    </collision>
  </link>

  <joint name="world_to_table" type="fixed">
      <parent link="world"/>
      <child link="table"/>
      <origin xyz="0 0 0.0" rpy="0 0 0"/>
  </joint>

  <joint name="ee_to_k_mount" type="fixed">
      <parent link="ee_link"/>
      <child link="k_mount"/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 ${pi/2} 0.0"/>
  </joint>

  <joint name="k_mount_to_k" type="fixed">
      <parent link="k_mount"/>
      <child link="k4a"/>
      <origin xyz="0.0525 0.0 0.0" rpy="${pi} ${pi} ${pi/2}"/>
  </joint>

  <joint name="table_to_robot_mount" type="fixed">
      <parent link="table"/>
      <child link="robot_mount"/>
      <origin xyz="0 0 0.012" rpy="0 0 0"/>
  </joint>

  <joint name="robot_mount_to_robot" type="fixed">
      <parent link="robot_mount"/>
      <child link="base_link"/>
      <origin xyz="0 0 0.007" rpy="0 0 0"/>
  </joint>

  <joint name="k_mount_to_adapter" type="fixed">
      <parent link="k_mount"/>
      <child link="robotiq_arg2f_base_link"/>
      <origin xyz="0 0 0.02" rpy="0 0 0"/>
  </joint>

  <joint name="winml_to_k4a_optical" type="fixed">
      <parent link="k4a_link_optical"/>
      <child link="winml_link"/>
      <origin xyz="0 0 0.027" rpy="0 0 0"/>
  </joint>

  <joint name="k4a_optical_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="k4a"/>
    <child link="k4a_link_optical"/>
  </joint>

</robot>
